Support vector parameter for MultiAppCommandLineControl (#17120)
idaholab/moose: modules/stochastic_tools/doc/content/source/controls/MultiAppCommandLineControl.md

!listing multiapps/commandline_control/master_multiple.i block=Controls

## Vector Parameter

> The vector parameter can be altered statistically with `MultiAppCommandLineControl`. To illustrate its usage, we consider an input file listed below:

Reviewer: ```suggestion
A vector parameter can be altered statistically with `MultiAppCommandLineControl`. For example, consider the input file listed below:
```

> !listing multiapps/batch_commandline_control/master_vector.i block=Controls
> 
> In this input file, the `param_names` includes a vector parameter with 4 entries called `Materials/const/prop_values` and two scalar parameters called `Mesh/xmax` and `Mesh/ymax`.
> 
> Several cases are discussed as follows:
> 
> 1. All four entries will be altered: set `param_names = Materials/const/prop_values[0,1,2,3]`. The `[0,1,2,3]` is the global column index of the provided distributions which implies that each entry corresponds to a different distribution.  
> 
> 2. The third and fourth entry will be altered with a same distribution: set `param_names = Materials/const/prop_values[0,1,2,2]`. The repeated index "2" means that their values are the same and will be sampled from a same distribution.
> 
> 3. The second entry will not be altered: set `param_names = Materials/const/prop_values[0,(0.5),1,2]`. In this case, the second entry will be set as 0.5 while the first, third and last entries will be altered statistically. In general, a constant value will be provided inside the parentheses bracket.
> 
> 4. If `[]` is used, it must be provided to every parameter. By default, if the ``[]`` is not provided it is assumed that values are scalar and captured in order.
> 

Reviewer: ```suggestion
!listing multiapps/batch_commandline_control/master_vector.i block=Controls

In this input file, the `param_names` includes a vector parameter with 4 entries called `Materials/const/prop_values` and two scalar parameters called `Mesh/xmax` and `Mesh/ymax`.

Several cases exist for modifying the vector parameter:

1. All four entries will be altered: set `param_names = Materials/const/prop_values[0,1,2,3]`. The `[0,1,2,3]` is the global column index of the provided distributions which implies that each entry corresponds to a different distribution.  

2. The third and fourth entry will be altered with a same distribution: set `param_names = Materials/const/prop_values[0,1,2,2]`. The repeated index "2" means that their values are the same and will be sampled from a same distribution.

3. The second entry will not be altered: set `param_names = Materials/const/prop_values[0,(0.5),1,2]`. In this case, the second entry will be set as 0.5 while the first, third and last entries will be altered statistically. In general, a constant value will be provided inside the parentheses bracket.

!alert info
If `[]` is used, it must be provided to every parameter. By default, if the `[]` is not provided it is assumed that values are scalar and captured in order.

```
